src/services/textChanges.ts

```diff
@@ -165,7 +181,7 @@ namespace ts.textChanges {
             // fullstart
             // when b is replaced - we usually want to keep the leading trvia
             // when b is deleted - we delete it
-            return position === Position.Start ? start : fullStart;
+            return startPosition === LeadingTriviaOption.IncludeAll ? fullStart : start;

```

Comment:
`fullStart` includes the trailing trivia of the preceding token?  I thought it was just the leading trivia of this token (though it's confusing and I may be misremembering).

Reply:
If the token `b` has a preceding token `a`, `b`'s `fullStart` will be the position after `a`'s `end`. So from what I understand, `a`'s trailing trivia (if any) will occur starting on `b`'s `fullStart` and end somewhere before `b`'s `start` (depending on the occurrence of line breaks).

Reply:
Sounds plausible.  I've forgotten how this works multiple times. :smile: